contentcuration/contentcuration/db/models/manager.py

```diff
@@ -75,35 +99,39 @@ def _attempt_lock(self, tree_ids, values):
                     .values(*values)
                 )
             yield
+            log_lock_time_spent(time.time() - start)
 
     @contextlib.contextmanager
     def lock_mptt(self, *tree_ids):
         # If this is not inside the context of a delay context manager
         # or updates are not disabled set a lock on the tree_ids.
         if not self.model._mptt_is_tracking and self.model._mptt_updates_enabled:
             tree_ids = sorted((t for t in set(tree_ids) if t is not None))
-            # Lock based on MPTT columns for updates on any of the tree_ids specified
-            # until the end of this transaction
-            mptt_opts = self.model._mptt_meta
-            values = (
-                mptt_opts.tree_id_attr,
-                mptt_opts.left_attr,
-                mptt_opts.right_attr,
-                mptt_opts.level_attr,
-                mptt_opts.parent_attr,
-            )
-            try:
-                with self._attempt_lock(tree_ids, values):
-                    yield
-            except OperationalError as e:
-                if "deadlock detected" in e.args[0]:
-                    logging.error(
-                        "Deadlock detected while trying to lock ContentNode trees for mptt operations, retrying"
-                    )
+            if tree_ids:
+                # Lock based on MPTT columns for updates on any of the tree_ids specified
+                # until the end of this transaction
+                mptt_opts = self.model._mptt_meta
+                values = (
+                    mptt_opts.tree_id_attr,
+                    mptt_opts.left_attr,
+                    mptt_opts.right_attr,
+                    mptt_opts.level_attr,
+                    mptt_opts.parent_attr,
+                )
+                try:
                     with self._attempt_lock(tree_ids, values):
                         yield
-                else:
-                    raise
+                except OperationalError as e:
+                    if "deadlock detected" in e.args[0]:
+                        logging.error(
+                            "Deadlock detected while trying to lock ContentNode trees for mptt operations, retrying"
+                        )
+                        with self._attempt_lock(tree_ids, values):
+                            yield
+                    else:
+                        raise
+            else:
+                yield
```

Comment:
I try to avoid nesting code too much if I can get around it a little. Although I don't think that's necessarily the python-way. I'm curious of your thoughts? For instance, inverting the checks and doing a `yield` and `return` would make the entire function a bit flatter.

Reply:
I think we could move the tree_ids check outside of the if statement, as it's not a huge problem to do the tree_id filtering and setting.